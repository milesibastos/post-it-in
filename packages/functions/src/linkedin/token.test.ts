import axios from "axios";
import handler from "./token";

jest.mock("axios");

const access_token = "#access_token";

const post = axios.post as unknown as jest.Mock;

post.mockResolvedValue({data: {access_token}});

// Exchange Authorization Code for an Access Token
// https://docs.microsoft.com/en-us/linkedin/shared/authentication/authorization-code-flow?context=linkedin/consumer/context#step-3-exchange-authorization-code-for-an-access-token
test("should send code to request token", async () => {
  const config = {
    headers: {
      "Content-Type": "application/x-www-form-urlencoded",
    },
  };
  const token = await handler("xyz");
  expect(post)
      .toBeCalledWith(
          expect.any(String),
          expect.objectContaining({code: "xyz"}),
          config
      );

  expect(token).toMatchSnapshot();
});
