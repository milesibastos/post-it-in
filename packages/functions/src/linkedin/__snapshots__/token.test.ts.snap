// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should send code to request token 1`] = `
Object {
  "access_token": "#access_token",
}
`;
